runtime/lua/vim/lsp/util.lua

```diff
@@ -135,7 +135,7 @@ function M.apply_text_document_edit(text_document_edit)
   local text_document = text_document_edit.textDocument
   local bufnr = vim.uri_to_bufnr(text_document.uri)
   -- TODO(ashkan) check this is correct.
-  if api.nvim_buf_get_changedtick(bufnr) > text_document.version then
+  if M.buf_versions[bufnr] > text_document.version then
```

Comment:
This can produce nils

I added the following:
`M.buf_versions[bufnr] ~= nil and`

at the beginning of the if, and with that change and this pr, I no longer get this error when trying to rename stuff with gopls